frontend/src/App.tsx

```diff
@@ -1,20 +1,73 @@
+import { useRecoilCallback, useRecoilValue, useResetRecoilState, useSetRecoilState } from "recoil";
 import { UploadPage } from "./pages/upload/UploadPage";
 import { LibraryPage } from "./pages/library/LibraryPage";
 import { NotFound } from "./components/NotFound/NotFound";
-import { modalDialogAtom } from "./store/atoms";
+import {
+  colorsCounterAtom,
+  compoundIdSelectedAtom,
+  excelFileAtom,
+  experimentsTableSizeAtom,
+  modalDialogAtom,
+  projectSelectedIdAtom,
+  selectedSubExperimentsColorAtom,
+  selectedSubExperimentsIdsAtom,
+  targetIdSelectedAtom,
+} from "./store/atoms";
 import { CreateMethodologyDialog } from "./components/dialogs/CreateMethodologyDialog";
 import { CreateProjectDialog } from "./components/dialogs/CreateProjectDialog";
 import { MethodologyDescriptionDialog } from "./components/dialogs/MethodologyDescriptionDialog";
 import { DeleteMethodologyDialog } from "./components/dialogs/DeleteMethodologyDialog";
 import { AddLinkDialog } from "./components/dialogs/AddLinkDialog";
+import { DashboardPage } from "./pages/dashboard/DashboardPage";
+import { chartColors } from "./store/types";
+import { SimpleDialog } from "./components/dialogs/SimpleDialog";
 
 export const App: FunctionComponent = (): ReactElement => {
   const modalDialog = useRecoilValue(modalDialogAtom);
+  const location = useLocation();
+  const resetExcelFile = useResetRecoilState(excelFileAtom);
+  const resetProjectSelectedId = useResetRecoilState(projectSelectedIdAtom);
+  const resetCompoundIdSelected = useResetRecoilState(compoundIdSelectedAtom);
+  const resetTargetIdSelected = useResetRecoilState(targetIdSelectedAtom);
+  const resetExperimentsTableSizeAtom = useResetRecoilState(experimentsTableSizeAtom);
+  const setSelectedSubExperimentsIds = useSetRecoilState(selectedSubExperimentsIdsAtom);
+  const selectedSubExperiments = useRecoilValue(selectedSubExperimentsIdsAtom);
+
+  const resetColorsCounter = useRecoilCallback(({ reset }) => () => {
+    for (let i = 0; i < chartColors.length; i++) {
+      reset(colorsCounterAtom(chartColors[i]));
+    }
+  });
+
+  const resetSubexperimentsColors = useRecoilCallback(({ reset }) => () => {
+    const arr = Array.from(selectedSubExperiments);
+    for (let i = 0; i < arr.length; i++) {
+      reset(selectedSubExperimentsColorAtom(arr[i]));
+    }
+  });
+
+  useEffect(() => {
+    // Cleaners of state when we leave a page
+    if (location.pathname !== "/upload") {
+      resetExcelFile();
+    }
+
+    if (location.pathname !== "/dashboard") {
+      resetColorsCounter();
+      resetSubexperimentsColors();
+      resetProjectSelectedId();
+      resetCompoundIdSelected();
+      resetTargetIdSelected();
+      resetExperimentsTableSizeAtom();
+      // For some weird reason reset doesn't update selectedExperimentsQuery selector
+      setSelectedSubExperimentsIds(new Set<number>());
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [location.pathname]);
```

Comment:
>Route has attribute render

and what? how can replace useEffect? it's an equivalent of children of Route. how it's helpful and replacement of useEffect? you should learn react-router better, perpahs :putya:

>why don't you load all needed information for pages here?
because there are plenty of things to finish, and I don't consider it's 100% better: you might have ton of experiments, and when you go on the website to just look at projects, you will be obliged to load all this shit. also, if you want to see information about several experiments, when you have 1000 of them, in your suggestion, I need to upload ton of irrelevant bullshit. So there are a lot of drawback in such approach and I didn't have time for pieces of information which might be reasonable o pre-fetch. Also, I would rather fix updating states: for now there are places where all the data (a list) if refetched when one item of list is updated.